### ISSUE TITLE
EventBridge: aws_cloudwatch_event_target being duplicated

### ISSUE THREAD

**Issue by abonawit** [hashicorp/terraform-provider-aws#16394]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v0.12.18
provider.aws v3.16.0

### Affected Resource(s)

* aws_cloudwatch_event_target

### Terraform Configuration Files

```hcl

resource "aws_cloudwatch_event_target" "my-test-target" {
  target_id = "my-test-target"
    rule = aws_cloudwatch_event_rule.my-test-rule.name
    arn = "arn:aws:lambda:xx-xxxx-x:xxxxxxxxxxxxxx:function:my-test-lambda"

    event_bus_name = "my-custom-bus"
}
```

### Debug Output

```hcl
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # module.instances.module.my-test-rules.aws_cloudwatch_event_target.my-test-target will be created
  + resource "aws_cloudwatch_event_target" "my-test-target" {
      + arn            = "arn:aws:lambda:xx-xxxx-x:xxxxxxxxxxxxxx:function:my-test-lambda"
      + event_bus_name = "my-custom-bus"
      + id             = (known after apply)
      + rule           = "my-test-rule"
      + target_id      = "my-test-target"
    }
```

### Expected Behavior
No new targets would be created since they were already created in the rule on a previous terraform apply.

### Actual Behavior

A new target was created in the rule even though no new changes were made to the target in the terraform.

### Steps to Reproduce

1. Create eventbridge rule via terraform
2. Add target to rule via terraform
3. Plan and apply rule and target
4. Run "terraform init" again.
5. Run "terraform plan" without making any changes to the eventbridge rule target
6. Output will show a new target resource to create
7. Terraform apply adds new duplicate target while leaving existing target untouched.

### Important Factoids

This behavior occurred in both scenarios where target_id was populated, and where it was omitted.

**Comment by bflad** [comment]
Hi @abonawit 👋  Thank you for opening this and sorry you ran into trouble here.

In your reproduction steps you mention running `terraform init` again. Was the `aws_cloudwatch_event_target` resource originally applied with version 3.14.0 of the Terraform AWS Provider from November 06, 2020?

**Comment by abonawit** [comment]
Hi @bflad yes the original issue was found in the 3.14.0 version. I had changed the version before posting the issue and forgot to change it back.

**Comment by bflad** [comment]
Thank you for that additional information. We should be able to add a patch on top of the resource logic that is causing this issue. In the meantime, if you need to plan/apply your configuration on the newer version, you should be able to use `terraform state rm` and `terraform import` to workaround the issue. If you need assistance with that process, please reach out.

**Comment by abonawit** [comment]
Thanks @bflad. Any estimation on time frame for the patch?

**Comment by bflad** [comment]
Fix submitted: https://github.com/hashicorp/terraform-provider-aws/pull/16484

**Comment by bflad** [comment]
The fix for the state upgrade has been merged and will release with version 3.20.0 of the Terraform AWS Provider. 👍

**Comment by abonawit** [comment]
Thank you @bflad for your quick work on this! Very much appreciated.
